### PR TITLE
feat:fallBackOnMiss for parsers

### PR DESCRIPTION
**Pull request by khareyash05**
## Related Issue
FallBackOnMiss for parsers

Closes: #1839 

#### Describe the changes you've made
Adds a feature to check if FallBackOnMiss is disabled -> return empty buffer else allow passThrough

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, local variables)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### CODE REVIEW

File: pkg/core/proxy/integrations/generic/decode.go [hunk 1]
```diff
@@ -71,6 +71,17 @@ func decodeGeneric(ctx context.Context, logger *zap.Logger, reqBuf []byte, clien
...
+					_, err := clientConn.Write(([]byte{}))
+					if err != nil {
+						utils.LogError(logger, err, "failed to write empty response to the client")
+						errCh <- err
+						return
+					}
+					errCh <- nil
+					return
+				}
+
```
Comment (gouravkrosx) [inline pkg/core/proxy/integrations/generic/decode.go:84]: Also add an info log that `no mock matched with the current request, hence connecting to the real service` along with the destination address in all the parsers.

**Comment by gouravkrosx** [comment]
@khareyash05 And can you please also add HTTP response with msg: `no mock found for the request` instead of returning nothing in case of HTTP parser.

**Comment by khareyash05** [comment]
image

**Comment by gouravkrosx** [comment]
Closing as the related issue is not relevant now.

**Pull request closed without merging**